File: 1523-count-odd-numbers-in-an-interval-range.java
class CountOddNumbersInAnIntervalRange {
    public int countOdds(int low, int high) {
        int length = high - low + 1;
        int res = length / 2;
        if (length % 2 == 1 && low % 2 == 1) {
            res++;
        }

        return res;
    }
  
    public static void main(String[] args) {
        CountOddNumbersInAnIntervalRange sol = new CountOddNumbersInAnIntervalRange();
        System.out.println("Output: " + sol.countOdds(3, 7) + ", expected: 3");
        System.out.println("Output: " + sol.countOdds(8, 10) + ", expected: 1");
        System.out.println("Output: " + sol.countOdds(21, 22) + ", expected: 1");
    }
}

File: 1523-count-odd-numbers-in-an-interval-range.py
class Solution:
    def countOdds(self, low: int, high: int) -> int:
        '''
        We get the total count of numbers from low to high and divide by 2. If low is odd, add 1 to the res
        '''
        length = high - low + 1
        res = length // 2
        if low % 2 and length % 2:
            res += 1

        return res


sol = Solution()
print(f'output: {sol.countOdds(3, 7)}, expected: 3')
print(f'output: {sol.countOdds(8, 10)}, expected: 1')
print(f'output: {sol.countOdds(21, 22)}, expected: 1')
